druid/src/widget/view_switcher.rs

```diff
@@ -79,9 +79,7 @@ impl<T: Data, U: PartialEq> Widget<T> for ViewSwitcher<T, U> {
             self.active_child = Some(WidgetPod::new((self.child_builder)(&child_id, data, env)));
             self.active_child_id = Some(child_id);
             ctx.children_changed();
-        }
-
-        if let Some(child) = self.active_child.as_mut() {
+        } else if let Some(child) = self.active_child.as_mut() {
```

Comment:
If I remember correctly from zulip, there's some slightly tricky interaction with lifecycle, right? Maybe add a comment.

Reply:
Can do

Reply:
This has ended up in someone elses PR anyway